streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -539,6 +540,10 @@ public <K1> KStream<K1, V> toStream(final KeyValueMapper<? super K, ? super V, ?
 
     @Override
     public KTable<K, V> suppress(final Suppressed<? super K> suppressed) {
+        if (graphNode.isOutputVersioned().isPresent() && graphNode.isOutputVersioned().get()) {
```

Comment:
This eager check might not be necessary, and only works if the upstream node is materialized. We could also remove it, but it might give a better stacktrace if we have it.

Otherwise, the error only surfaced from `build()` was is harder to reason about.

Reply:
Makes sense. Could be good to leave a comment in the code so future readers know that this check is incomplete, and that the one in InternalStreamsBuilder is the "real" one.

Should we also update the other error message (in InternalStreamsBuilder) to make clear to users that versioned-ness can be inherited downstream, since they will only hit that exception if that's the case? Don't think it's necessary, just a thought.

EDIT: just saw your other comment below with the same question haha.